osf/migrations/0170_populate_file_versions_through.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2019-03-03 17:52
+from __future__ import unicode_literals
+
+import logging
+
+from django.db import migrations, connection
+
+logger = logging.getLogger(__file__)
+
+
+def restore_default_through_table(state, schema):
+    sql = """
+        DROP TABLE osf_basefilenode_versions;
+        CREATE TABLE osf_basefilenode_versions AS
+        SELECT
+            new_thru.basefilenode_id,
+            new_thru.fileversion_id
+        FROM
+            osf_basefileversionsthrough AS new_thru;
+
+        ALTER TABLE osf_basefilenode_versions ADD COLUMN id SERIAL PRIMARY KEY;
+        ALTER TABLE osf_basefilenode_versions ADD CONSTRAINT osf_basefilenod_basefilenode_id_b0knah27_fk_osf_basefilenode_id FOREIGN KEY (basefilenode_id) REFERENCES osf_basefilenode DEFERRABLE INITIALLY DEFERRED;
+        ALTER TABLE osf_basefilenode_versions ALTER COLUMN basefilenode_id
+        SET
+            DATA TYPE INTEGER;
+        ALTER TABLE osf_basefilenode_versions ALTER COLUMN fileversion_id
+        SET
+            NOT NULL;
+        ALTER TABLE osf_basefilenode_versions ALTER COLUMN fileversion_id
+        SET
+            DATA TYPE INTEGER;
+        ALTER TABLE osf_basefilenode_versions ALTER COLUMN basefilenode_id
+        SET
+            NOT NULL;
+        ALTER TABLE osf_basefilenode_versions ADD CONSTRAINT osf_basefilenode__fileversion_id_93etanfc_fk_osf_fileversion_id FOREIGN KEY (fileversion_id) REFERENCES osf_fileversion DEFERRABLE INITIALLY DEFERRED;
```

Comment:
I may need to dynamically create this constraint name?

Reply:
For this one, can you just use ALTER_TABLE to rename the new table to be the old name and drop the version_name column (and maybe any unnecessary indices, if that doesn't happen automatically)?

Reply:
The migration before this which adds the osf_basefileversionsthrough table (which we delete in 0170 and build our own), is expecting that table to be there to revert the migration.

Reply:
Would it be easy to just re-run 0169 to generate the table fresh? Or maybe just include 0170 in 0169 even if that's not usually how we do it?

Reply:
Maybe, I can explore that option.  I've played around with removing 0169 altogether, but then I get weird errors if it's not there.  I'd prefer 0169 to not be there at all, as that seems a waste, but maybe there's something else that 0170 needs to take care of.